code/modules/mob/say.dm

```diff
@@ -145,6 +145,29 @@
 
 	return null
 
+/**
+  * Replaces all brackets from the input with italics
+  *
+  * Requires an equal number of opening and closing brackets, to avoid overflow.
+  */
+/mob/proc/say_emphasis(input)
+	/// Number of opening brackets in the input
+	var/opening = 0
+	/// Number of closing brackets in the input
+	var/closing = 0
+
+	for(var/I in 1 to length(input))
+		var/char = input[I]
+		if(char == "(")
+			opening++
+		else if(char == ")")
+			closing++
+
+	if(opening <= closing) // Opening number isn't higher
+		input = replacetext(input, "(", "<i>")
```

Comment:
I'm a little paranoid this allows someone to type ))))))(( and italicise an entire window by leaving an unterminated <I> span. Please test this.

Reply:
Yeah this logic in general doesn't quite make sense. You count the opening and closing brackets to make sure you can't leave a hanging open tag, but at the same time you don't actually check for hanging open tags. You can just do `)(` to defeat this magic.

Just use a regex. 

The PR from 2017 @AffectedArc07 linked earlier in the comments here had nothing to do with OOC's frequency of use, but was a result of BYOND initializing regexes on server startup improperly, as is explained in the PR and the BYOND forum thread linked there. My bet is this will work fine today, you can always revisit the issue if you have consistent issues with regex usage.

Also, for the love of god, too many comments.

Reply:
That was an issue, yes. Although it would only effect the rest of the `say` line so it wouldn't have been too catastrophic.

Fixed now though, and a few other things I noticed. Fingers crossed I got them all.